### PR TITLE
Added a Test For Corner Kick Scenario And Made RatePass Work For It

### PR DESCRIPTION
**Pull request by garethellis0**
### Description
Added a test for one-time pass/shot in a corner kick scenario where the robot we're passing to is marked by an enemy

### Testing Done
Unit test.

### Resolved Issues
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/util/parameter/config/ai.yaml [hunk 1]
```diff
@@ -83,6 +83,14 @@ AI:
         Maximum time into the future at which the pass should occur. This gives the
         upper bound on the pass start time, relative to the current time. This is in
         seconds.
+    enemy_intercept_delay:
```
Comment (MathewMacDougall) [inline src/thunderbots/software/util/parameter/config/ai.yaml]: I think "reaction time" would be better here, and it matches what's in the code.

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: src/thunderbots/software/util/parameter/config/ai.yaml (hunk 1 above)
Comment (garethellis0) [reply src/thunderbots/software/util/parameter/config/ai.yaml]: Great point, changed.

**Review by MathewMacDougall — DISMISSED** [review]
LGTM, just need CI to pass

**Review by MathewMacDougall — APPROVED** [review]
LGTM

**Pull request merged**